### PR TITLE
CLI | Update the Cli Calls to SCA Risk Management Service to use Export Service (AST-42958)

### PR DESCRIPTION
**Pull request by AlvoBen**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Change deprecated risk-management API to export new API .

### References

> https://checkmarx.atlassian.net/browse/AST-42958

### Testing

> Run tests locally

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: internal/commands/result.go [hunk 1]
```diff
@@ -2273,52 +2313,85 @@ func addPackageInformation(
...
+}
+
+func updateDependencyPaths(dependencyPaths [][]wrappers.DependencyPath, locationsByID map[string][]*string) {
+	for i := range dependencyPaths {
+		head := &dependencyPaths[i][0]
+		head.Locations = locationsByID[head.ID]
+		head.SupportsQuickFix = len(dependencyPaths[i]) == 1
+
+		for _, location := range locationsByID[head.ID] {
+			head.SupportsQuickFix = head.SupportsQuickFix && util.IsPackageFileSupported(*location)
```
Comment (tamarleviCm) [inline internal/commands/result.go:2383]: what about this line: 
currentPackage.SupportsQuickFix = currentPackage.SupportsQuickFix || head.SupportsQuickFix

File: internal/commands/result.go [hunk 2]
```diff
@@ -2273,52 +2313,85 @@ func addPackageInformation(
...
+		}
+	}
+}
+
+func buildFixLink(result *wrappers.ScanResult) string {
+	if result.ID != "" {
+		return "https://devhub.checkmarx.com/cve-details/" + result.ID
```
Comment (tamarleviCm) [inline internal/commands/result.go:2390]: save it as const

File: internal/services/export.go [hunk 3]
```diff
@@ -1,23 +1,65 @@
...
+	}
+	logger.PrintIfVerbose(fmt.Sprintf("Initiated export request for scanID %s. ExportID: %s", scanID, exportID))
+
+	exportResponse, err := pollForCompletion(exportWrapper, exportID.ExportID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if exportResponse != nil && exportResponse.ExportStatus == "Completed" && exportResponse.FileURL != "" {
```
Comment (tamarleviCm) [inline internal/services/export.go:53]: save "Completed" as const

File: internal/commands/result.go [hunk 4]
```diff
@@ -2273,52 +2313,85 @@ func addPackageInformation(
...
-						currentPackage.TypeOfDependency = directDependencyType
-					} else {
-						currentPackage.TypeOfDependency = indirectDependencyType
-					}
-					result.ScanResultData.ScaPackageCollection = &currentPackage
-					break
-				}
-			}
+		if result.Type == commonParams.ScaType {
+			processResult(result, locationsByID, typesByCVE, scaPackageMap)
```
[the remark this answers is not in the document]
Comment (AlvoBen) [reply internal/commands/result.go:2322]: Can be multiple results

File: internal/commands/result.go (hunk 1 above)
Comment (AlvoBen) [reply internal/commands/result.go:2397]: Missed it by mistake. Added it to updatePackages func

File: internal/commands/result.go [hunk 5]
```diff
@@ -2273,52 +2314,98 @@ func addPackageInformation(
...
+	scaPackageMap map[string]wrappers.ScaPackageCollection,
+	locationsByID map[string][]*string,
+	currentID string,
+) {
+	packages, found := scaPackageMap[currentID]
+	if !found {
+		return
+	}
+
+	updateDependencyPaths(packages.DependencyPathArray, locationsByID)
```
Comment (tamarleviCm) [inline internal/commands/result.go:2357]: suggestion:
return the SupportsQuickFix from the updateDependencyPaths method.

**Review by tamarleviCm — APPROVED** [review]

**Pull request merged**